locobot/agent/locobot_mover.py

```diff
@@ -57,6 +68,78 @@ def __init__(self, ip=None, backend="locobot", use_dslam=False):
         self.uv_one_in_cam = np.dot(intrinsic_mat_inv, uv_one)
         self.backend = backend
     
+    def check(self):
+        """
+        Sanity checks all the mover interfaces. 
+        
+        Checks move by moving the locobot around in a square and reporting L1 drift and total time taken
+            for the three movement modes available to the locobot - using PyRobot slam (vslam), 
+            using Droidlet slam (dslam) and without using any slam (default)
+        Checks look and point by poiting and looking at the same target.
+        """
+        self.reset_camera()
+        table = PrettyTable(["Command", "L1 Drift", "Time (sec)"])
+        sq_table = PrettyTable(["Mode", "Total L1 drift", "Total time (sec)"])
+        
+        def l1_drift(a, b):
+            return round(abs(a[0] - b[0]) + abs(a[1] - b[1]), ndigits=3)
+
+        def execute_move(init_pos, dest_pos, cmd_text, use_map=False, use_dslam=False):
+            logging.info("Executing {} ... ".format(cmd_text))
+            start = time.time()
+            self.move_absolute([dest_pos], use_map=use_map, use_dslam=use_dslam)
+            end = time.time()
+            tt = round((end-start), ndigits=3)
+            pos_after = self.get_base_pos_in_canonical_coords()
+            drift = l1_drift(pos_after, dest_pos)
+            logging.info("Finished Executing. \nDrift: {} Time taken: {}".format(drift, tt))
+            table.add_row([cmd_text, drift, tt])
+            return drift, tt
+        
+        def move_in_a_square(magic_text, side=0.3, use_vslam=False, use_dslam=False):
+            """
+            Moves the locobot in a square starting from the bottom right - goes left, forward, right, back.
+
+            Args:
+                magic_text (str): unique text to differentiate each scenario
+                side (float): side of the square
+
+            Returns:
+                total L1 drift, total time taken to move around the square.
+            """
+            pos = self.get_base_pos_in_canonical_coords()
+            logging.info("Initial agent pos {}".format(pos))
+            dl, tl = execute_move(pos, [pos[0]-side, pos[1], pos[2]], "Move Left " + magic_text, use_map=use_vslam, use_dslam=use_dslam)
+            df, tf = execute_move(pos, [pos[0]-side, pos[1]+side, pos[2]], "Move Forward " + magic_text, use_map=use_vslam, use_dslam=use_dslam)
```

Comment:
should we update the `pos` variable after performing every `execute_move` command ?

Reply:
Thought about this when I was writing it but decided against it because that scenario tests something much simpler - just traversing a straight line. In the current framing, the task is being able to traverse the entire square. Even if one of the move commands is wrong, the next move command should still try and go to the right next vertex of the square. I prefer that setup just for the greater complexity it captures. Let me know if you feel otherwise strongly though.

Reply:
This sounds good as well, however then should we take into account only the final error `db` instead of `dl+df+dr+db` each execution error?

Reply:
No - that would hide any in-path errors, which is useful to know especially to track improvement in locomotion over time.

Reply:
K, that sounds good